modules/search/zlibrary.py

```diff
@@ -0,0 +1,109 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from bs4 import BeautifulSoup
+
+# Using default empty strings for query, lang, ext to avoid &ext=None etc.
+meta = {
+	'name': 'Zlibrary Search',
+	'author': 'Vikas Kundu',
+	'version': '0.1',
+	'description': 'Search books and articles on Zlibrary . ',
+	'sources': ('Zlibrary',),
+	'options': (
+		('query', '', True, 'Query string to search for', '-q', 'store', str),
+		('exact', False, False, 'Search for exact query(True or False, default=False)', '-e', 'store_true', bool),
+		('start_year', 0, False, 'Year to start search from(min: 1800 onwards)', '-sy', 'store', int),
+		('end_year', 0, False, 'Year till which to search for(max: present year)', '-ey', 'store', int),
+		('lang', '', False, 'Language in which to search(i.e. English)', '-ln', 'store', str),
+		('ext', '', False, 'Extension of the ebook or article(i.e. pdf, epub, txt, rar, mobi)', '-ex', 'store', str),
+		('count', 50, False, 'No. of results to show(default=50)', '-c', 'store', int) # 50 entries per page
+	),
+    'examples': ('zlibrary -q <QUERY> -ex pdf',)
```

Comment:
-q ... --ext/-e pdf

Reply:
can't do that, exact option is already -e

Reply:
bool options should not have a short option. so change -e to --exact

Reply:
Leaving bool option empty causes parser errors. So should i change its short option to -exact ?

Reply:
So should i change its short option to -exact? I mean it